packages/badge/index.tsx

```diff
@@ -1,23 +1,47 @@
-import type { BadgeProps } from './types'
-import productStyles from '@hashicorp/platform-product-meta/style.module.css'
 import classNames from 'classnames'
+import { BadgeProps } from './types'
+import useProductMeta from '@hashicorp/platform-product-meta'
 import s from './style.module.css'
 
-export default function Badge({
-  children,
+const Badge = ({
+  text,
   variant = 'primary',
-  theme = 'neutral',
-}: BadgeProps) {
-  let themeClass
-  const productClass = productStyles[theme]
-  if (productClass) {
-    themeClass = classNames(s.product, productClass)
-  } else {
-    themeClass = s[theme]
-  }
+  theme = 'vagrant',
+  page = 'light',
+}: BadgeProps) => {
+  const products = [
+    'boundary',
+    'consul',
+    'nomad',
+    'packer',
+    'terraform',
+    'vault',
+    'vagrant',
+    'waypoint',
+  ]
+  const isProductTheme = products.includes(theme)
+  const productName = isProductTheme ? theme : undefined
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore - The ternary above ensures that `useProductMeta` always gets
+  // the appropriate value but this does not seem to appease the the TS gods
```

Comment:
A bit stumped by this TS error. Would love feedback so as to avoid `ts-ignore`'ing.